### PR TITLE
Add basic Persian(Farsi) language support + two small models

### PR DESCRIPTION
**Pull request by JKhakpour**
## Description
Hi,
I added some features to Persian language class. This includes a lemmatizer and some other stuffs. I have used [HAZM](https://github.com/sobhe/hazm/) and [petit](https://github.com/JKhakpour/petit) packages (both MIT License) and [Seraji's Universal Dependencies dataset](https://github.com/UniversalDependencies/UD_Persian-Seraji) (Creative Commons v4). 

I have also trained a model for this package. Three datasets are used to train this model:
- [fastText pre-trained models on wikipedia](https://github.com/facebookresearch/fastText/blob/master/pretrained-vectors.md) (reduced to 50K vectors)
- [UD_Persian-Seraji](https://github.com/UniversalDependencies/UD_Persian-Seraji) to train tagger and dependency parser
- [ArmanPersianNER](https://github.com/HaniehP/PersianNER) for named entity recognition

The latter dataset is "Academic only". As I'm not into "Academic Only" datasets, I prepared two models for Persian language, one including ner and one without ner:
- [fa_core_wiki_sm](https://www.dropbox.com/s/wg6n6fbqidesp9i/fa_core_wiki_sm.zip?dl=0)  (Tagger + Parser + NER)
- [fa_xcore_wiki_sm](https://www.dropbox.com/s/hvqzr9mfnfmjbpn/fa_xcore_wiki_sm.zip?dl=0)  (Tagger + Parser)

## Checklist

- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by honnibal** [comment]
This looks really good, thanks!

What sort of accuracy did you get for the dependency parser and tagger? Were you able to run the standard Universal Dependencies evaluations?

Our process for language models is to make sure we have our own scripts which train the model, so that if we make breaking changes, we're always able to regenerate the model. So, we won't be able to immediately host your model file, but it should be easy to add to our scripts, so that Persian can receive full support :tada:

**Comment by JKhakpour** [comment]
Hi @honnibal 
There is no modifications on the dataset. These are steps I did to create dataset:
1. I initialized model from fastText's  pre-trained vectors (reducing to 50000)
```
python -m spacy init-model fa fa_core_wiki_sm --vectors-loc wiki.fa.vec.new --prune-vectors 50000
```
2. converted UD_Seraji and trained tagger and parser based on this data
```
 python -m spacy convert UD_Persian-Seraji/fa_seraji-ud-dev.conllu udfa-json
 python -m spacy convert UD_Persian-Seraji/fa_seraji-ud-train.conllu udfa-json
 python -m spacy convert UD_Persian-Seraji/fa_seraji-ud-test.conllu udfa-json
python -m spacy train fa models udfa-json/fa_seraji-ud-train.json  udfa-json/fa_seraji-ud-dev.json -N -v ./fa_core_wiki_sm
```
this is the output:
```
➜ python -m spacy train fa models udfa-json/fa_seraji-ud-train.json  udfa-json/fa_seraji-ud-dev.json -N -v ./fa_core_wiki_sm
dropout_from = 0.2 by default
dropout_to = 0.2 by default
dropout_decay = 0.0 by default
batch_from = 1 by default
batch_to = 16 by default
batch_compound = 1.001 by default
max_doc_len = 5000 by default
Load vectors model ./fa_core_wiki_core
beam_width = 1 by default
beam_density = 0.0 by default
learn_rate = 0.001 by default
optimizer_B1 = 0.9 by default
optimizer_B2 = 0.999 by default
optimizer_eps = 1e-08 by default
L2_penalty = 1e-06 by default
grad_norm_clip = 1.0 by default
embed_size = 7000 by default
token_vector_width = 128 by default
parser_hidden_depth = 1 by default
parser_maxout_pieces = 2 by default
token_vector_width = 128 by default
hidden_width = 200 by default
embed_size = 7000 by default
history_feats = 0 by default
history_width = 0 by default
Itn.	P.Loss	N.Loss	UAS	NER P.	NER R.	NER F.	Tag %	Token %
0       26304.492       0.000   83.520  0.000   0.000   0.000   96.267  100.000 13205.9 0.0                                                                                                                   
1       1126.918        0.000   85.279  0.000   0.000   0.000   96.684  100.000 14112.7 0.0                                                                                                                   
2       385.620 0.000   85.923  0.000   0.000   0.000   96.747  100.000 14021.4 0.0                                                                                                                           
3       246.457 0.000   86.292  0.000   0.000   0.000   96.703  100.000 14100.1 0.0                                                                                                                           
....                                                                                                                          
28      54.756  0.000   85.909  0.000   0.000   0.000   96.551  100.000 13969.4 0.0                                                                                                                           
29      52.748  0.000   85.850  0.000   0.000   0.000   96.551  100.000 13923.8 0.0                                                                                                                           
Saving model...
``` 
3. And trained a ner using ArmanPersoNer corpus. I wrote some python code to add ner to this pipeline. The code is written in an ipython notebook and it's a little messy. I will try to clean it up and copy it to my gists in next few days. I will insert the link here.

**Comment by honnibal** [comment]
@JKhakpour Thanks!

When you do the `convert` step, I think you'll need to set `-n 10`, to merge some sentences together. Otherwise the model won't work properly on running text, as it'll expect only one sentence per document.

**Comment by JKhakpour** [comment]
I trained both models again. You can find them in below links:
- [fa_wiki_core_sm](https://www.dropbox.com/s/alo1hxw8srwk2o0/fa_wiki_core_sm.zip?dl=0)
- [fa_wiki_xcore_sm](https://www.dropbox.com/s/9smy5n8dlntglmj/fa_wiki_xcore_sm.zip?dl=0)

And this is the python code I used to train ner:
[Spacy Train NER on ArmanPersoNer.ipynb](https://gist.github.com/JKhakpour/a8447d61c1eeb891cf4b028234fcf7c0)

It's not a good code, but when I tried to clean the code, it had thrown some errors which I was not able to fix. SO, I'm publishing it as is and I will try to change and fix the code later.

**Comment by JKhakpour** [comment]
@honnibal should I change anything on these models/codes? Is there anything needs to fix before merge?

**Comment by honnibal** [comment]
@JKhakpour Looks good! Thanks again; merging!
